### PR TITLE
Fix/cypress #165 junit xml reports

### PR DESCRIPTION
**Pull request by herrkoch**
Quickstarter runs fine locally and on Jenkins; not yet tested with release manager (provisioning of the release manager failed in my playground)

### CODE REVIEW

**Comment by clemensutschig** [comment]
@herrkoch whats the error you got for the rm?

**Comment by herrkoch** [comment]
Hi @clemensutschig I get this error when provisioning the rm on my playground ("daniel" namespace on our OpenShift): `org.jenkinsci.plugins.scriptsecurity.sandbox.RejectedAccessException: Scripts not permitted to use method groovy.lang.GroovyObject getProperty java.lang.String (org.ods.quickstarter.Context.bitbucketUrl)`
I am only generating one JUnit xml file with the results of all Cypress e2e tests and store it at the expected location (as there is no differentiation between acceptance / integration / installation in the Cypress quickstarter). Should we introduce these 3 separate "stages" also in Cypress?

**Comment by metmajer** [comment]
@herrkoch Hi Daniel and thanks for your contribution. The release manager expects three files, as otherwise, it cannot distinguish between these test types. Depending on the type, the test results need to go into different documents. Can you please fix that? Best you take a look at the spock/geb quickstarter and implement it as similarly as possible. Cheers!

**Comment by herrkoch** [comment]
Thanks for the explanation @metmajer , I think I understand now what has to be done. My idea to solve this is to execute `cypress run` 3 times in a row, each time with a different config for `integrationFolder` and `mochaFile` for installation / acceptance / integration. First I'll follow up with one of the BI X colleagues on the release manager deployment issue.

**Comment by metmajer** [comment]
Most welcome, @herrkoch. The issue you are describing above happens if you access a property of an object that does not exist. Are you using a version of ODS that does not have `org.ods.quickstarter.Context.bitbucketUrl`? Is it written in the correct way?

**Comment by clemensutschig** [comment]
Context.bitbucketHost.. ;-) misleading jenkins error

**Comment by herrkoch** [comment]
Hi @metmajer @clemensutschig I adapted the quickstarter to now execute e2e tests in 3 stages (acceptance + installation + integration), each one generating a JUnit XML result file that is being stashed by Jenkins. @michaelsauter helped me to fix the deployment of the release manager in my playground (daniel-cd) and achieved a successful run with the cypress quickstarter configured as dependency.

**Comment by herrkoch** [comment]
It's worth noting that in the cypress quickstarter execution on Jenkins this happens as well:
`Stashing testResults from location: 'build/test-results/test'
[Pipeline] sh (Moving test results to system location: build/test-results/test)
+ mkdir -p build/test-results/test build/test-results/test
+ cp -rf 'build/test-results/test/*' build/test-results/test
+ true
cp: cannot stat 'build/test-results/test/*': No such file or directory
[Pipeline] sh (Counting test results in build/test-results/test)
+ wc -l
+ ls -la 'build/test-results/test/*.xml'
ls: cannot access build/test-results/test/*.xml: No such file or directory
[Pipeline] echo
No xUnit results found!!`
Not sure where this comes from but if I put the result files to `build/test-results/test` it will be stashed "automatically" but not picked up by the release manager. I had to include the same lines of code in the Jenkinsfile.template as for the spock/geb quickstarter to properly stash the results files so they are picked up by the release manager.

**Comment by clemensutschig** [comment]
All good this Magic happens somewhere else.. no need to do this.  :-) Check
the spock qs.. there you see the stashing..

Daniel Koch <notifications@github.com> schrieb am Fr., 29. Mai 2020, 18:41:

File: e2e-cypress/Jenkinsfile.template [hunk 1]
```diff
@@ -1,9 +1,9 @@
 // See https://www.opendevstack.org/ods-documentation/ for usage and customization.
 
-@Library('ods-jenkins-shared-library@@ods_git_ref@') _
+@Library('ods-jenkins-shared-library@3.x') _
 
 odsComponentPipeline(
-  imageStreamTag: '@ods_namespace@/jenkins-slave-nodejs10-angular:@ods_image_tag@',
+  imageStreamTag: 'ods/jenkins-slave-nodejs10-angular:3.x',
```
Comment (michaelsauter) [inline e2e-cypress/Jenkinsfile.template]: You need to leave the variables in - that should always be replaced when the quickstarter is provisioned with the versions that the user chose.

Comment (herrkoch) [reply e2e-cypress/Jenkinsfile.template]: Hi Michael, yes this was an accidental copy/paste change that I already fixed last Friday directly after my first commit (I guess that's why GitHub says "Outdated").

Comment (michaelsauter) [reply e2e-cypress/Jenkinsfile.template]: Sorry, didn't realise. It showed that to me as part of the diff. Now if I go there it isn't there anymore ...

Comment (herrkoch) [reply e2e-cypress/Jenkinsfile.template]: Yeah, I guess GitHub send you an email with a direct link to the changes of the first commit. They should change that and make it a link to the changes of the full PR. :-)

**Comment by herrkoch** [comment]
Regarding the stashing: I copied that part over from the Jenkinsfile of the spock/geb quickstarter (with some path adjustments) and the stashing/unstashing worked fine in my test run with the release manager.

**Pull request merged**